frame/transaction-payment/asset-tx-payment/src/lib.rs

```diff
@@ -223,6 +228,8 @@ where
 		Self::AccountId,
 		// imbalance resulting from withdrawing the fee
 		InitialPayment<T>,
+		// asset_id for the transaction payment
+		Option<ChargeAssetIdOf<T>>,
```

Comment:
I am very confused why this PR is introducing this.

It makes sense, but it should have existed already. There is no way that adding an event should require adding more data into a signed extension

Reply:
I add this inorder to get 'asset_id' ,otherwise need to add parameter 'self' into the interface 'post_dispatch' in my humble opinion. Maybe there is a other better solution.

Reply:
Hi , @shawntabrizi  What do you think about this now? Would you like to share your solution ? 😃

Reply:
@shawntabrizi that is not adding any new data to the signed extension. The only thing this is doing is that it adds a new return value to `pre_dispatch` that is then passed to `post_dispatch`

Reply:
Right, @bkchr you explain it more clearly . :+1: